tests/RobotFramework/tests/tedge/apt_plugin.robot

```diff
@@ -0,0 +1,69 @@
+...    No support is provided by tedge config so store apt.name and apt.maintainer.
+
+Resource            ../../resources/common.resource
+Library             ThinEdgeIO
+
+Suite Setup         Setup
+Suite Teardown      Get Logs
+
+
+*** Test Cases ***
+
+
+tedge-apt-plugin list
+    ${apt_list}    Execute Command    /etc/tedge/sm-plugins/apt list
+    Should Contain    ${apt_list}    apt
+    # Should Contain    ${apt_list}    wget
+    Should Contain    ${apt_list}    dpkg
+
+tedge-apt-plugin list --name
+    ${apt_list_name}    Execute Command    /etc/tedge/sm-plugins/apt list --name '(c8y-).*'
+    Should Contain    ${apt_list_name}    c8y-configuration-plugin
+    Should Contain    ${apt_list_name}    c8y-firmware-plugin
+    Should Contain    ${apt_list_name}    c8y-log-plugin
+    Should Contain    ${apt_list_name}    c8y-remote-access-plugin
+    Should Not Contain    ${apt_list_name}    tedge
+    Should Not Contain    ${apt_list_name}    tedge-agent
+    Should Not Contain    ${apt_list_name}    tedge-apt-plugin
+    Should Not Contain    ${apt_list_name}    tedge-mapper
+    Should Not Contain    ${apt_list_name}    tedge-watchdog
+
+
+tedge-apt-plugin list --maintainer
+    ${apt_workaround}    Set Variable    apt    
+    ${apt_list_maintainer}    Execute Command    /etc/tedge/sm-plugins/apt list --maintainer '.*(thin-edge.io).*'
+    Should Contain    ${apt_list_maintainer}    c8y-configuration-plugin
+    Should Contain    ${apt_list_maintainer}    c8y-firmware-plugin
+    Should Contain    ${apt_list_maintainer}    c8y-log-plugin
+    Should Contain    ${apt_list_maintainer}    c8y-remote-access-plugin
+    Should Contain    ${apt_list_maintainer}    tedge
+    Should Contain    ${apt_list_maintainer}    tedge-agent
+    Should Contain    ${apt_list_maintainer}    tedge-apt-plugin
+    Should Contain    ${apt_list_maintainer}    tedge-mapper
+    Should Contain    ${apt_list_maintainer}    tedge-watchdog
+    Should Not Contain    ${apt_list_maintainer}    ${apt_workaround}\t
+    Should Not Contain    ${apt_list_maintainer}    dpkg
+
+tedge-apt-plugin list --name --maintainer
+    ${apt_list_name_maintainer}    Execute Command    /etc/tedge/sm-plugins/apt list --name '(c8y-).*' --maintainer '.*(APT Development Team).*'
+    Should Contain    ${apt_list_name_maintainer}    c8y-configuration-plugin
+    Should Contain    ${apt_list_name_maintainer}    c8y-firmware-plugin
+    Should Contain    ${apt_list_name_maintainer}    c8y-log-plugin
+    Should Contain    ${apt_list_name_maintainer}    c8y-remote-access-plugin
+    Should Contain    ${apt_list_name_maintainer}    apt
+    Should Not Contain    ${apt_list_name_maintainer}    tedge
+    Should Not Contain    ${apt_list_name_maintainer}    tedge-agent
+    Should Not Contain    ${apt_list_name_maintainer}    tedge-apt-plugin
+    Should Not Contain    ${apt_list_name_maintainer}    tedge-mapper
+    Should Not Contain    ${apt_list_name_maintainer}    tedge-watchdog
+
```

Comment:
We're missing a test case to validate the `software.plugin.max_packages` functionality...for example the setting used to prevent large lists. When set, then the returned list of packages should be less or equal to the `software.plugin.max_packages` setting

Reply:
Beware that this max number of packages doesn't apply directly on the plugin themselves. This is done *only* when sending software list to the cloud.

`/etc/tedge/sm-plugins/apt list` returns the complete list of packages. Only the `head -n $max_packages` will be forwarded to the cloud.

Reply:
Testing the maximum number published on the cloud can be done as the `tedge-agent` level.

Indeed, `tedge-agent` publishes the list of installed packages on request.

- Sending a request: `tedge mqtt pub tedge/commands/req/software/list '{"id":"test-request-software-list"}'`
- The list is published over mqtt on `tedge/commands/req/software/list`.